compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -657,6 +657,15 @@ public void visit(BLangSimpleVariable variable) {
             BType type = variable.typeNode.getBType();
             recordGlobalVariableReferenceRelationship(Types.getReferredType(type).tsymbol);
         }
+
+        if (isGlobalVarSymbol(symbol)) {
+            BLangExpression expr = variable.expr;
+            if (expr instanceof BLangInvocation) {
```

Comment:
Here, This can be a group expression or compound expression, (i.e. `a + foo()` ) etc. We need to handle this case as well.